app/test/test_ftp.py

```diff
@@ -32,6 +33,48 @@ def tearDown(self):
         shutil.rmtree(self.root)
 
 
+class ExceptionTests(unittest.TestCase):
+
+    params = {
+        "user": "testuser",
+        "password": "password",
+        "host": "0.0.0.0",
+        "port": 2121,
+        "working_directory": ".",
+    }
+
+    @unittest.mock.patch("ftpclient.FTP.connect", side_effect=Exception("Connect failure"))
+    def test_error_on_connect(self, connect_mock):
+        worker = FTPWorker(**self.params)
+        with worker as broker:
+            self.assertFalse(broker)
+            connect_mock.assert_called_once_with("0.0.0.0", 2121, timeout=30)
```

Comment:
This is causing the Codacy failure "bind all interfaces" - does it have to be mocked to `0.0.0.0` or can it be something else to stop Codacy complaining?